src/btree/bt_page.c

```diff
@@ -418,49 +478,63 @@ __page_merge_deltas_common_merge_loop(WT_SESSION_IMPL *session, WT_CELL_UNPACK_A
                 min_delta = NULL;
             }
         } else {
-            /* New implementation: build disk image */
+            /* New implementation: build_disk == true: append to new_image */
             if (cmp < 0) {
+                /* Base entry wins */
                 /*
                  * !!! NOTE: The following commented code is a placeholder for the actual
                  * implementation of base_key = &base[i++]; base_val = &base[i++];
                  */
                 if (row_leaf_page) {
                     /* Pack row-leaf base key/value. */
-                } else if (internal_page) {
+                } else if (row_internal_page) {
                     /* Pack internal base key/value. */
+                    WT_RET(__wt_rec_pack_internal_key_addr(
+                      session, new_image, &base[i], &base[i + 1], false, &p_ptr));
+
+                    entry_count += 2;   /* key + value cells */
+                    final_entries += 1; /* one ref (child) emitted */
                 }
-                /*
-                 * !!! NOTE: The following commented code is a placeholder for the actual
-                 * implementation of
-                 * __wti_rec_kv_copy(session, p, base_key);
-                 * p += base_key->len;
-                 * __wti_rec_kv_copy(session, p, base_val);
-                 * p += base_val->len;
-                 */
-                i++;
+                i += 2;
             } else {
                 if (row_leaf_page) {
                     /* Pack row-leaf delta entry. */
-                } else if (internal_page && !__wt_delta_cell_type_visible_all(min_delta)) {
+                } else if (row_internal_page && !__wt_delta_cell_type_visible_all(min_delta)) {
                     /* Pack internal delta entry. */
+                    WT_RET(__wt_rec_pack_internal_key_addr(
+                      session, new_image, min_delta, min_delta, true, &p_ptr));
+                    entry_count += 2;   /* key + value */
+                    final_entries += 1; /* one ref (child) emitted */
                 }
-                /*
-                 * !!! NOTE: The following commented code is a placeholder for the actual
-                 * implementation of
-                 * __wti_rec_kv_copy(session, p, min_delta->key);
-                 * p += min_delta->key.len;
-                 * __wti_rec_kv_copy(session, p, min_delta->value);
-                 * p += min_delta->value.len;
-                 */
                 if (cmp == 0)
                     i += 2;
                 delta_idx[min_d]++;
                 /* We consumed this delta, so recompute next round */
                 min_delta = NULL;
             }
+        }
+    }
 
-            final_entries++;
+    if (build_disk) {
+        /* Finalize header once after all appends -- recompute hdr from
```

Comment:
comment format

Reply:
Fixed in efccce3b67416ff6057b1cd4e9008bd4b17e1ba4